ColorPatterns/Helpers/Starling.swift

```diff
@@ -0,0 +1,266 @@
+        }
+    }
+
+    // MARK: - Public API (Loading Sounds)
+
+    public func load(resource: String, type: String, for identifier: SoundIdentifier) {
+        DispatchQueue.global(qos: .utility).async { [weak self] in
+            if let blockSelf = self {
+                if let url = Bundle.main.url(forResource: resource, withExtension: type) {
+                    blockSelf.load(sound: url, for: identifier)
+                } else {
+                    blockSelf.handleNonFatalError(StarlingError.resourceNotFound(name: "\(resource).\(type)"))
+                }
+            }
+        }
+    }
+
+    public func load(sound url: URL, for identifier: SoundIdentifier) {
+        if let file = try? AVAudioFile(forReading: url) {
+            didFinishLoadingAudioFile(file, identifier: identifier)
+        } else {
+            handleNonFatalError(StarlingError.audioLoadingFailure)
+        }
+    }
+
+    // MARK: - Public API (Playback)
+
+    public func play(_ sound: SoundIdentifier, allowOverlap: Bool = true) {
+        DispatchQueue.global(qos: .userInitiated).async { [weak self] in
+            self?.performSoundPlayback(sound, allowOverlap: allowOverlap)
+        }
+    }
+
+    // MARK: - Internal Functions
+
+    private func performSoundPlayback(_ sound: SoundIdentifier, allowOverlap: Bool) {
+        // Note: self is used as the lock pointer here to avoid
+        // the possibility of locking on _swiftEmptyDictionaryStorage
+        objc_sync_enter(self)
+        let file = files[sound]
+        objc_sync_exit(self)
+
+        guard let audio = file else {
+            handleNonFatalError(StarlingError.invalidSoundIdentifier(name: sound))
+            return
+        }
+
+        func performPlaybackOnFirstAvailablePlayer() {
+            guard let player = firstAvailablePlayer() else { return }
+
+            objc_sync_enter(players)
+            player.play(audio, identifier: sound)
+            objc_sync_exit(players)
+        }
+
+        if allowOverlap {
+           performPlaybackOnFirstAvailablePlayer()
+        } else {
+            if !soundIsCurrentlyPlaying(sound) {
+                performPlaybackOnFirstAvailablePlayer()
+            }
+        }
+    }
+
+    private func soundIsCurrentlyPlaying(_ sound: SoundIdentifier) -> Bool {
+        objc_sync_enter(players)
+        defer { objc_sync_exit(players) }
+        // TODO: This O(n) loop could be eliminated by simply keeping a playback tally
+        for player in players {
+            let state = player.state
+            if state.status != .idle && state.sound == sound {
+                return true
+            }
+        }
+        return false
+    }
+
+    private func firstAvailablePlayer() -> StarlingAudioPlayer? {
+        objc_sync_enter(players)
+        defer { objc_sync_exit(players) }
+        let player: StarlingAudioPlayer? = {
+            // TODO: A better solution would be to actively manage a pool of available player references
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/vandermesis/ColorPatterns/blob/8c9d8b0cbd258e51fec07cc9cd8e5c65cf9e26ea/ColorPatterns/Helpers/Starling.swift#L156"></span>TODOs should be resolved (A better solution would be to …).<br />
<code>todo</code> <code>Starling.swift:156</code></td>
    </tr>
  </tbody>
</table>